fre/app/mask_atmos_plevel/mask_atmos_plevel.py

```diff
@@ -0,0 +1,156 @@
+    # After the masking is done, change the attribute to True.
+    # Note: these are string types called True and False, not boolean types.
+    for var in list(ds_in.variables):
+        if 'pressure_mask' in ds_in[var].attrs:
+            if ds_in[var].attrs['pressure_mask'].lower() == 'false':
+                ds_out[var] = mask_field_above_surface_pressure(ds_in, var, ds_ps)
+                ds_out[var].attrs['pressure_mask'] = "True"
+                fre_logger.info(f"Finished processing '{var}' and set 'pressure_mask' to True")
+            else:
+                fre_logger.debug(f"Not processing '{var}' (because 'pressure_mask' is not False)")
+        else:
+                fre_logger.debug(f"Not processing '{var}' (because it does not have 'pressure_mask'")
+
+    # Write the output file if anything was done
+    if ds_out.variables:
+        fre_logger.info(f"Modifed {list(ds_out.variables)} variables, so writing into new file '{outfile}'")
+        write_dataset(ds_out, ds_in, outfile)
+    else:
+        fre_logger.debug(f"No variables modified, so not writing output file '{outfile}'")
+
+
+def mask_field_above_surface_pressure(ds: xr.Dataset, var: str, ds_ps: xr.Dataset) -> xr.Dataset:
+    """
+    Mask data with pressure larger than surface pressure
+    :param ds: Input dataset to be masked
+    :type infile: xarray.Dataset
+    :param var: Input variable to be masked
+    :type var: str
+    :return:Output masked dataset
+    :rtype: xrray.Dataset
+
+    .. note:: Missing values are set to 1.0e20.
+    """
+
+    # retrieve the pressure coordinate variable
+    plev = pressure_coordinate(ds, var)
+
+    # broadcast pressure coordinate and surface pressure to
+    # the dimensions of the variable to mask
+    plev_extended, _ = xr.broadcast(plev, ds[var])
+    ps_extended, _ = xr.broadcast(ds_ps["ps"], ds[var])
+
+    # masking do not need looping
+    masked = xr.where(plev_extended > ps_extended, 1.0e20, ds[var])
+
+    # copy attributes, but it doesn't include the missing values
+    attrs = ds[var].attrs.copy()
+
+    # add the missing values back
+    attrs['missing_value'] = 1.0e20
+    attrs['_FillValue'] = 1.0e20
+    masked.attrs = attrs
+
+    # transpose dims like the original array
+    masked = masked.transpose(*ds[var].dims)
+    fre_logger.debug(f"Processed {var}")
+
+    return masked
+
+
+def pressure_coordinate(ds: xr.Dataset, varname: str) -> xr.DataArray:
+    """
+    Return the pressure coordinate of the Dataset or None if
+    the Dataset does not have a pressure coordinate.
+    :param ds: Input dataset to inspect
+    :type ds: xarray.Dataset
+    :param var: Input variable name to inspect
+    :type var: str
+    :return: Pressure coordinate variable
+    :rtype xarray.DataArray
+
+    .. warning:: Returns None if no pressure coordinate variable can be found
+    """
+
+    pressure_coord = None
+
+    for dim in list(ds[varname].dims):
```

Comment:
I'm not sure if .dims is being deprecated ,and .sizes is being recommended

Reply:
I don't think .dims is being deprecated. There's no warning and the documentation still has it-

https://docs.xarray.dev/en/stable/generated/xarray.DataArray.dims.html#xarray.DataArray.dims

Reply:
We're using DataArray.dims here which seems OK to me.

Though, I believe you Dana that I could be doing something non-compliant here. If I am please point it out again, please.